src/base/emu-i386/kvm.c

```diff
@@ -1021,6 +1080,59 @@ static unsigned int kvm_run(void)
       break;
     case KVM_EXIT_MMIO:
       /* for ROM: simply ignore the write and continue */
+      if (memcheck_is_rom(run->mmio.phys_addr))
+	break;
+
+      /* from the KVM api.txt: "the corresponding operations are complete
+	 (and guest state is consistent) only after userspace has re-entered
+	 the kernel with KVM_RUN. The kernel side will first finish
+	 incomplete operations and then check for pending signals." */
+#ifdef KVM_CAP_IMMEDIATE_EXIT
+      if (kvm_cap_immediate_exit)
+	run->immediate_exit = 1;
+      else
+#endif
+	pthread_kill(pthread_self(), KVM_IMMEDIATE_EXIT_SIG);
```

Comment:
Can we not have that? :)
If immediate_exit works, why to care?
At least please make is a separate patch,
for an easy revert in the future.

Reply:
Yes, good idea, will do that (the separate patch).

It's needed on kernels 4.12 and older (think enterprise distros such as RHEL7, everything else has something newer).

Reply:
Maybe under !KVM_CAP_IMMEDIATE_EXIT
at least?

Reply:
I put almost all the compat code together and in a separate patch now.